python/foglamp/services/south/ingest.py

```diff
@@ -291,7 +301,11 @@ def increment_discarded_readings(cls):
 
         while list_index <= cls._max_concurrent_readings_inserts-1:
             if cls._stop:
-                break  # Terminate this method
+                readings = 0
+                for i, v in cls._sensor_stats.items():
+                    readings += v
+                if cls._readings_stats+cls._discarded_readings_stats+readings == 0:
```

Comment:
add space around `+` operator and add a comment why this SUM should be 0 to break